plugins/turing/config/config.go

```diff
@@ -1,22 +1,23 @@
 package config
 
 import (
+	"fmt"
 	"sync"
 
-	"github.com/caraml-dev/turing/engines/experiment/pkg/request"
 	"github.com/go-playground/validator/v10"
 )
 
-// RequestParameter captures a single parameter's config for parsing the incoming
-// request and extracting values
-type RequestParameter struct {
-	// Parameter specifies the name of the parameter, expected by the XP API
-	Parameter string `json:"parameter" validate:"required"`
-	// Field specifies the name of the field in the incoming request header/payload
-	Field string `json:"field" validate:"required"`
-	// FieldSrc specifies whether the field is located in the request header or payload
-	FieldSrc request.FieldSource `json:"field_source" validate:"required,oneof=header payload"`
-}
+type FieldSource string
+
+const (
+	// PayloadFieldSource is used to represent the request payload
+	PayloadFieldSource FieldSource = "payload"
+	// HeaderFieldSource is used to represent the request header
+	HeaderFieldSource FieldSource = "header"
+	// NoneFieldSource is used to represent that there is no field source,
+	// i.e., the variable has not been configured.
+	NoneFieldSource FieldSource = "none"
```

Comment:
Adding `NoneFieldSource` for the scenario where the user wants to skip configuring a variable.